lib/src/component_declaration/component_base.dart

```diff
@@ -700,17 +662,51 @@ abstract class _OverReactMapViewBase<K, V> {
   Map<K, V> get _map;
 }
 
+/// A class that mimics `MapBase` without all the additional functionality we don't need.
+///
+/// > Necessary for eventual upgrade to the Dart 2 SDK.
+abstract class _OverReactMapBase<K, V> extends Object implements MapViewMixin<K, V>, Map<K, V> {
+  // Manually implement members from `Map` that appear in the Dart 2 SDK
+  // without having to extend from `MapBase`.
```

Comment:
#nit I think this comment implies the wrong thing; we don't want to extend from `MapBase` because we want to just proxy that map.

Perhaps this class should be called `_MapProxy` or `_OverReactMapView`